### ISSUE TITLE
RFC: make tools menu stackable

### ISSUE THREAD

**Issue by pazos** [koreader/koreader#12647]
It is currently hardcoded as part of menusorter. For all the rest of menus that's ok because most meny entries are common between platforms and are not plugin dependant.

On the tools menu there might be a bunch of things if you have a bunch of plugins enabled. It can also be almost empty, with all relevant entries buried into submenus.

Proposal: make tools menu stackable.

- Keep an array of suggested/featured tools
- Keep an array of all tools
- Fill menus based on arrays
    - Fill the menu with
    ```
    manage plugins
    manage patches (if patches present)
    ------------------
    featured or all
    ------------------
    more
    ```

    - Fill **more** submenu with:
    ```
    advanced settings
    developer options
    ------------------
    rest of featured and all
    ```

I know I can change that with menusorter itself and that's wonderful. I do like menusorter a lot :)

Let me ask for an exception in the menusorter :)
We have a menu here with potentially too many entries and user cases. It deserves special treatment.

**Comment by Frenzie** [comment]
I think it might help if the problem description were a bit clearer. What exactly are you trying to solve?

**Comment by pazos** [comment]
> I think it might help if the problem description were a bit clearer. What exactly are you trying to solve?

1. The tools menu might be almost empty and, even then, the entries might be buried on nested submenus.

2. There's a limitation on when the tools are placed, so if they belong to "more tools" they won't be shown on top menu, even when there's plenty of space available.

**Comment by Frenzie** [comment]
Right, and if it's sufficiently empty "more tools" isn't even needed at all. I'm not convinced this is much of a problem: if you disable so many plugins you can just make a custom menu order for those 2 menus.

Key for me is the ability to put the stuff from `tools` and `more tools` anywhere at will.

It might also be worth keeping in mind that I'm not attached to anything that just so happens to be in tools remaining there.

But if it bothers you some kind of virtual menu that populates tools and overflows into more tools is fine by me, although I am mildly concerned about added maintenance complexity.

**Comment by ummm288** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.
